src/mlpack/bindings/R/print_type_doc_impl.hpp

```diff
@@ -24,29 +24,29 @@ namespace r {
 template<typename T>
 std::string PrintTypeDoc(
     util::ParamData& data,
-    const typename std::enable_if<!arma::is_arma_type<T>::value>::type*,
-    const typename std::enable_if<!util::IsStdVector<T>::value>::type*,
-    const typename std::enable_if<!data::HasSerialize<T>::value>::type*,
-    const typename std::enable_if<!std::is_same<T,
-        std::tuple<data::DatasetInfo, arma::mat>>::value>::type*)
+    const std::enable_if_t<!arma::is_arma_type<T>::value>*,
+    const std::enable_if_t<!util::IsStdVector<T>::value>*,
+    const std::enable_if_t<!data::HasSerialize<T>::value>*,
+    const std::enable_if_t<!std::is_same_v<T,
+        std::tuple<data::DatasetInfo, arma::mat>>>*)
 {
   // A flag type.
-  if (std::is_same<T, bool>::value)
+  if (std::is_same_v<T, bool>)
   {
     return "A boolean flag option (i.e. `TRUE` or `FALSE`).";
   }
   // An integer.
-  else if (std::is_same<T, int>::value)
+  else if (std::is_same_v<T, int>)
   {
     return "An integer (i.e., `1`).";
   }
   // A floating point value.
-  else if (std::is_same<T, double>::value)
+  else if (std::is_same_v<T, double>)
   {
     return "A floating-point number (i.e., `0.5`).";
   }
   // A string.
-  else if (std::is_same<T, std::string>::value)
+  else if (std::is_same_v<T, std::string>)
   {
     return "A character string (i.e., `\"hello\"`).";
```

Comment:
Partly. It can be made `constexpr`, but it shouldn't help much. The compiler is already now able to optimize the function to its full extend, the `if constexpr` would only help in case on of the branches wouldn't compile if `T` is not the desired type.
And I would like to keep this separated in order to not mix up multiple concerns in one pull request. I could add the `constexpr` in a follow up, if you like.

Reply:
There should be no need for marking it constexpr---the compiler should be able to recognize that this is a compile-time operation and optimize it accordingly. There are lots of other places in the code where we depend on compilers in exactly this way. Anyway I agree with @marlamb that we should avoid unrelated code changes in this PR and just stick with the changes related to the PR. :+1: